### PR TITLE
sarama: fix span name and add payload size sem conv

### PR DESCRIPTION
**Pull request by srikanthccv**
Updates the span name to follow messaging semantic conventions and adds message payload size attribute

### CODE REVIEW

**Review by hanyuancheung — APPROVED** [review]

**Review by XSAM — APPROVED** [review]
lgtm

File: instrumentation/github.com/Shopify/sarama/otelsarama/producer.go
Comment (Aneurysm9) [inline instrumentation/github.com/Shopify/sarama/otelsarama/producer.go:265]:
```suggestion
	size := producerMessageOverhead
	if version >= 2 {
		size = maximumRecordOverhead
		for _, h := range msg.Headers {
			size += len(h.Key) + len(h.Value) + 2*binary.MaxVarintLen32
		}
	}
```

Since `producerMessageOverhead` is a magic number only used here not sure whether it hurts to inline it into the `size` declaration. v0v

File: instrumentation/github.com/Shopify/sarama/otelsarama/producer.go
Comment (Aneurysm9) [inline instrumentation/github.com/Shopify/sarama/otelsarama/producer.go:250]: There are a lot of magic numbers in here and it's not clear why/when `binary.MaxVarintLen32` or `binary.MaxVarintLen64` are used.  Some comments would be helpful.

**Comment by dmathieu** [comment]
You mention you're updating the span name to follow semantic conventions.
Is there a spec specifying those conventions for kafka, or examples of other libraries using the new name?

**Comment by srikanthccv** [comment]
> Is there a spec specifying those conventions for kafka, or examples of other libraries using the new name?

I didn't lookup other libraries. There is no spec specifying those conventions for kafka or (any specific message system).

**Comment by dmathieu** [comment]
I'm not sure I understand how this is an update to "follow messaging semantic conventions" if no other libraries is doing the same thing, and no conventions are defined?

**Comment by srikanthccv** [comment]
>if no other libraries is doing the same thing

There is more likely chance other instrumentations are also not following the semconv so I didn't bother looking at them.

>and no conventions are defined?

I am not sure if they are going to add naming conventions for every messaging system. There is a general guideline which says span name should `<destination (topic)> <operation name>`. More [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md).

**Review by dmathieu — APPROVED** [review]

**Comment by srikanthccv** [comment]
I think this has enough approvals and comments are addressed @open-telemetry/go-instrumentaiton-maintainers Can someone merge this?

**Comment by srikanthccv** [comment]
Can this be merged so that It will be part of the next release?  I would like to use this feature.

**Pull request merged**
